systest/cluster/discover.go

```diff
@@ -7,14 +7,53 @@ import (
 	"strings"
 
 	"golang.org/x/sync/errgroup"
-	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	v1 "k8s.io/api/core/v1"
+	apimetav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 
 	"github.com/spacemeshos/go-spacemesh/systest/testcontext"
 )
 
+func discoverPoets(ctx *testcontext.Context) ([]*v1.Pod, error) {
```

Comment:
i think it is possible to implement it without so much duplication

Reply:
used *NodeClient for poet too.